test/plugin/base_test.rb

```diff
@@ -2154,18 +2265,25 @@ def log_entry(i)
     "test log entry #{i}"
   end
 
-  def check_labels(labels, expected_labels)
-    return if labels.empty? && expected_labels.empty?
-    labels.each do |key, value|
-      assert value.is_a?(String), "Value #{value} for label #{key} " \
-        'is not a string: ' + value.class.name
-      assert expected_labels.key?(key), "Unexpected label #{key} => #{value}"
-      assert_equal expected_labels[key], value, 'Value mismatch - expected ' \
-        "#{expected_labels[key]} in #{key} => #{value}"
+  # If check_exact_labels is true, assert 'labels' and 'expected_labels' match
+  # exactly. If check_exact_labels is false, assert 'labels' is a subset of
+  # 'expected_labels'.
+  def check_labels(expected_labels, labels, check_exact_labels = true)
+    return if expected_labels.empty? && labels.empty?
+    expected_labels.each do |expected_key, expected_value|
```

Comment:
[Optional] Since the key is shared between the two hashes, you could rename `expected_key` to just `key`.

Reply:
The actual keys in the labels might be different from expected keys. I would prefer being explicit.